cargo doc fails

## What happens?
When I run `cargo doc --target riscv32imc-unknown-none-elf` on a project with `esp32c3-hal = "0.10.0"` as a dependency, it fails with the following error message.
```
    Checking esp-riscv-rt v0.3.0
 Documenting esp-riscv-rt v0.3.0
 Documenting esp-hal-procmacros v0.6.0
    Checking esp-hal-common v0.10.0
 Documenting esp-hal-common v0.10.0
error[E0428]: the name `InternalContext` is defined multiple times
  --> /home/polly/.cargo/registry/src/index.crates.io-6f17d22bba15001f/esp-hal-common-0.10.0/src/rom/md5.rs:67:1
   |
62 | struct InternalContext;
   | ----------------------- previous definition of the type `InternalContext` here
...
67 | struct InternalContext {
   | ^^^^^^^^^^^^^^^^^^^^^^ `InternalContext` redefined here
   |
   = note: `InternalContext` must be defined only once in the type namespace of this module

For more information about this error, try `rustc --explain E0428`.
error: could not document `esp-hal-common`

Caused by:
  process didn't exit successfully: `rustdoc --edition=2021 --crate-type lib --crate-name esp_hal_common /home/polly/.cargo/registry/src/index.crates.io-6f17d22bba15001f/esp-hal-common-0.10.0/src/lib.rs --cap-lints allow --target riscv32imc-unknown-none-elf -o /home/polly/Development/repro/target/riscv32imc-unknown-none-elf/doc --cfg 'feature="esp-riscv-rt"' --cfg 'feature="esp32c3"' --cfg 'feature="riscv"' --cfg 'feature="riscv-atomic-emulation-trap"' --cfg 'feature="rv-zero-rtc-bss"' --cfg 'feature="vectored"' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=143 -C metadata=b431a884e05003db -L dependency=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps -L dependency=/home/polly/Development/repro/target/debug/deps --extern bitflags=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libbitflags-0d8c316d089aab4b.rmeta --extern cfg_if=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libcfg_if-a7e189ade9f5712a.rmeta --extern critical_section=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libcritical_section-43ebbe3023cde884.rmeta --extern embedded_dma=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libembedded_dma-123a4fd9140cf194.rmeta --extern embedded_hal=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libembedded_hal-ccdb2116268097db.rmeta --extern procmacros=/home/polly/Development/repro/target/debug/deps/libesp_hal_procmacros-bff83fac9363a498.so --extern esp_riscv_rt=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libesp_riscv_rt-9bf1ca78d75d548c.rmeta --extern esp32c3=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libesp32c3-f838761dce6ca861.rmeta --extern fugit=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libfugit-3307809d5085c7d0.rmeta --extern log=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/liblog-4a2ae66dc17941ba.rmeta --extern nb=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libnb-28adfc6333f35e15.rmeta --extern paste=/home/polly/Development/repro/target/debug/deps/libpaste-d4a77623cdb6df0e.so --extern riscv_atomic_emulation_trap=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libriscv_atomic_emulation_trap-eed36d35e9d1aec8.rmeta --extern strum=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libstrum-191c54abdcd19d9d.rmeta --extern void=/home/polly/Development/repro/target/riscv32imc-unknown-none-elf/debug/deps/libvoid-98e3bce4932d8841.rmeta --crate-version 0.10.0 --cfg esp32c3 --cfg riscv --cfg single_core --cfg aes --cfg apb_ctrl --cfg apb_saradc --cfg assist_debug --cfg dma --cfg ds --cfg efuse --cfg extmem --cfg gpio --cfg gpio_sd --cfg hmac --cfg i2c0 --cfg i2s0 --cfg interrupt_core0 --cfg io_mux --cfg ledc --cfg rmt --cfg rng --cfg rsa --cfg rtc_cntl --cfg sensitive --cfg sha --cfg spi0 --cfg spi1 --cfg spi2 --cfg system --cfg systimer --cfg timg0 --cfg timg1 --cfg twai0 --cfg uart0 --cfg uart1 --cfg uhci0 --cfg uhci1 --cfg usb_device --cfg xts_aes --cfg adc --cfg assist_debug_sp_monitor --cfg assist_debug_region_monitor --cfg gdma --cfg radio --cfg phy --cfg bt --cfg wifi --cfg rom_crc_le --cfg rom_crc_be --cfg rom_md5_bsd --cfg pm_support_wifi_wakeup --cfg pm_support_bt_wakeup --cfg uart_support_wakeup_int --cfg gpio_support_deepsleep_wakeup` (exit status: 1)
warning: build failed, waiting for other jobs to finish...
```

I've read the other issues about docs, but they seem unrelated. It would be nice to have documentation locally :)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -4,7 +4,11 @@ on:
   pull_request:
     branches:
       - main
+    paths-ignore:
+      - "**/README.md"
   push:
+    paths-ignore:
+      - "**/README.md"
   workflow_dispatch:
 
 env:
@@ -21,7 +25,7 @@ concurrency:
 
 jobs:
   # --------------------------------------------------------------------------
-  # Check
+  # Check Packages
 
   esp-hal-smartled:
     runs-on: ubuntu-latest
@@ -53,6 +57,9 @@ jobs:
         run: cd esp-hal-smartled/ && cargo +esp check --features=esp32s2
       - name: check (esp32s3)
         run: cd esp-hal-smartled/ && cargo +esp check --features=esp32s3
+      # Ensure documentation can be built (requires a chip feature!)
+      - name: rustdoc
+        run: cd esp-hal-smartled/ && cargo doc --features=esp32c3
 
   esp32-hal:
     runs-on: ubuntu-latest
@@ -84,6 +91,9 @@ jobs:
         run: cd esp32-hal/ && cargo check --example=embassy_serial --features=embassy,embassy-time-timg0,async
       - name: check esp32-hal (async, i2c)
         run: cd esp32-hal/ && cargo check --example=embassy_i2c --features=embassy,embassy-time-timg0,async
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32-hal/ && cargo doc
 
   esp32c2-hal:
     runs-on: ubuntu-latest
@@ -123,6 +133,9 @@ jobs:
         run: cd esp32c2-hal/ && cargo +nightly check --example=embassy_i2c --features=embassy,embassy-time-systick,async
       - name: check esp32c2-hal (interrupt-preemption)
         run: cd esp32c2-hal/ && cargo check --example=interrupt_preemption --features=interrupt-preemption
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32c2-hal/ && cargo doc
 
   esp32c3-hal:
     runs-on: ubuntu-latest
@@ -164,6 +177,9 @@ jobs:
         run: cd esp32c3-hal/ && cargo +nightly check --example=embassy_i2c --features=embassy,embassy-time-systick,async
       - name: check esp32c3-hal (interrupt-preemption)
         run: cd esp32c3-hal/ && cargo check --example=interrupt_preemption --features=interrupt-preemption
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32c3-hal/ && cargo doc
 
   esp32c6-hal:
     runs-on: ubuntu-latest
@@ -203,6 +219,9 @@ jobs:
         run: cd esp32c6-hal/ && cargo +nightly check --example=embassy_i2c --features=embassy,embassy-time-systick,async
       - name: check esp32c6-hal (interrupt-preemption)
         run: cd esp32c6-hal/ && cargo check --example=interrupt_preemption --features=interrupt-preemption
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32c6-hal/ && cargo doc
 
   esp32h2-hal:
     runs-on: ubuntu-latest
@@ -242,6 +261,9 @@ jobs:
         run: cd esp32h2-hal/ && cargo +nightly check --example=embassy_i2c --features=embassy,embassy-time-systick,async
       - name: check esp32h2-hal (interrupt-preemption)
         run: cd esp32h2-hal/ && cargo check --example=interrupt_preemption --features=interrupt-preemption
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32h2-hal/ && cargo doc
 
   esp32s2-hal:
     runs-on: ubuntu-latest
@@ -276,6 +298,9 @@ jobs:
         run: cd esp32s2-hal/ && cargo check --example=embassy_serial --features=embassy,embassy-time-timg0,async
       - name: check esp32s2-hal (async, i2c)
         run: cd esp32s2-hal/ && cargo check --example=embassy_i2c --features=embassy,embassy-time-timg0,async
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32s2-hal/ && cargo doc
 
   esp32s3-hal:
     runs-on: ubuntu-latest
@@ -315,6 +340,9 @@ jobs:
         run: cd esp32s3-hal/ && cargo check --example=embassy_i2c --features=embassy,embassy-time-timg0,async
       - name: check esp32s3-hal (octal psram)
         run: cd esp32s3-hal/ && cargo check --example=octal_psram --features=opsram_2m --release # This example requires release!
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp32s3-hal/ && cargo doc
 
   esp-riscv-rt:
     runs-on: ubuntu-latest
@@ -332,6 +360,9 @@ jobs:
         run: cd esp-riscv-rt/ && cargo check --target=riscv32imc-unknown-none-elf -Zbuild-std=core
       - name: Check esp-riscv-rt (imac)
         run: cd esp-riscv-rt/ && cargo check --target=riscv32imac-unknown-none-elf -Zbuild-std=core
+      # Ensure documentation can be built
+      - name: rustdoc
+        run: cd esp-riscv-rt/ && cargo doc
 
   # --------------------------------------------------------------------------
   # MSRV

--- a/esp-hal-common/src/rom/md5.rs
+++ b/esp-hal-common/src/rom/md5.rs
@@ -58,9 +58,6 @@ use core::{
 // toml file then `InternalContext` will be either undefined or multiple
 // defined and this module will fail to compile letting you know to fix it
 
-#[cfg(doc)]
-struct InternalContext;
-
 #[cfg(rom_md5_bsd)]
 #[derive(Clone)]
 #[repr(C)]

